### PR TITLE
[cli] make linkmetrics cli as a separate module

### PR DESCRIPTION
**Pull request by Irving-cl**
The PR moves the cli for linkmetrics to a sub cli module to slim down
the `cli.cpp`.

### CODE REVIEW

File: src/cli/cli_link_metrics.hpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+#include "openthread-core-config.h"
+
+#include <openthread/link_metrics.h>
+
+#include "cli/cli_output.hpp"
+
+namespace ot {
+namespace Cli {
+
```
Comment (abtink) [inline src/cli/cli_link_metrics.hpp:45]: How about adding  `#if OPENTHREAD_CONFIG_MLE_LINK_METRICS_INITIATOR_ENABLE` here?

File: src/cli/cli_link_metrics.cpp [hunk 2]
```diff
@@ -0,0 +1,532 @@
...
+    if (blocking)
+    {
+        mLinkMetricsQueryInProgress = true;
+        error                       = OT_ERROR_PENDING;
+    }
+exit:
+    return error;
+}
+
+template <> otError LinkMetrics::Process<Cmd("mgmt")>(Arg aArgs[]) { return ProcessLinkMetricsMgmt(aArgs); }
```
Comment (abtink) [inline src/cli/cli_link_metrics.cpp:132]: We can consider removing `ProcessLinkMetricsMgmt()` and have its implementation directly here?

**Review by abtink — APPROVED** [review]
Looks great! Thanks. 
Couple of smaller suggestions below.

File: src/cli/CMakeLists.txt [hunk 3]
```diff
@@ -42,6 +42,7 @@ set(COMMON_SOURCES
     cli_dns.cpp
     cli_history.cpp
     cli_joiner.cpp
+    cli_link_metrics.cpp
```
Comment (jwhui) [inline src/cli/CMakeLists.txt:45]: Should we also update https://github.com/openthread/openthread/blob/main/src/cli/BUILD.gn ?

Comment (Irving-cl) [reply src/cli/CMakeLists.txt:45]: Updated.

**Review by Vyrastas — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**